### PR TITLE
feat: submit wearables

### PR DESCRIPTION
**Pull request by leanmendoza**
DRAFT: Waiting for builder api server works, currently the uploading of files isn't being save successfully in the S3 bucket.

# What? 
- [x] Create an identity to use with [builder-client](https://github.com/decentraland/builder-client) and linker-dapp
- [x] TODO: identity creation with this.wallet in `Decentraland` class.
- [x] Improve UX

# Why? 
https://github.com/decentraland/sdk/issues/157

### CODE REVIEW

File: src/commands/deploy/deploySmartWearable.ts [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+
+export default async function ({ dcl }: { dcl: Decentraland }) {
+  const project = dcl.workspace.getSingleProject()!
+  const assetJsonPath = path.resolve(
+    project.getProjectWorkingDir(),
+    'asset.json'
+  )
+  const assetJson = await readJSON(assetJsonPath)
+
+  console.log(chalk.bold('[BETA] This feature is a beta one.'))
```
Comment (menduz) [inline src/commands/deploy/deploySmartWearable.ts:27]:
```suggestion
  console.log(chalk.bold('[BETA] This is a beta feature.'))
```

File: src/commands/deploy/deploySmartWearable.ts [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+  }
+
+  spinner.create('Building smart wearable')
+
+  const assetBasicConfig = assetJson as sdk.AssetJson
+  const thumbnailContent = await readFile(
+    path.resolve(project.getProjectWorkingDir(), assetBasicConfig.thumbnail)
+  )
+
+  let originalFilesToIgnore = await project.getDCLIgnore()
```
Comment (gonpombo8) [inline src/commands/deploy/deploySmartWearable.ts:95]: already saw this logic 👀

File: src/commands/deploy/deploySmartWearable.ts [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  delete content['asset.json']
+
+  const item = new ItemFactory<Uint8Array>()
+    .newItem({
+      id: assetBasicConfig.id,
+      name: assetBasicConfig.name,
+      description: assetBasicConfig.description,
+      rarity: assetBasicConfig.rarity
+    })
+    .withCategory(assetBasicConfig.category as any)
```
Comment (gonpombo8) [inline src/commands/deploy/deploySmartWearable.ts:115]: why the category type mismatch ?

File: src/commands/deploy/deploySmartWearable.ts [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        entities: 0,
+        textures: 0
+      }
+    )
+
+  if (collectionId) {
+    item.withCollectionId(collectionId)
+  }
+
+  const builtItem = await item.build()
```
Comment (gonpombo8) [inline src/commands/deploy/deploySmartWearable.ts:135]: should we catch the item build error ?

File: src/commands/deploy/deploySmartWearable.ts [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+  spinner.succeed('Smart wearable built!')
+
+  dcl.on('link:ready', (url) => {
```
Comment (gonpombo8) [inline src/commands/deploy/deploySmartWearable.ts:139]: could we move this to a fn ? It's the same logic that we have at deployScene.
I think we should go for a fn approach and wrap all the common fns 

```suggestion
  dcl.on('link:ready', (url) => {
     openBrowser()
  })
```

File: src/lib/Decentraland.ts [hunk 6]
```diff
@@ -226,16 +232,44 @@ export class Decentraland extends EventEmitter {
...
-      return { signature, address }
+      return {
+        responseType: 'scene-deploy',
+        payload: { signature, address, chainId: ChainId.ETHEREUM_MAINNET }
+      }
+    }
+
+    const response = await this.link(messageToSign)
+    if (response.responseType !== 'scene-deploy') {
+      console.error({ response })
```
Comment (gonpombo8) [inline src/lib/Decentraland.ts:251]: should we remove this log ?

File: src/lib/Decentraland.ts [hunk 7]
```diff
@@ -226,16 +232,44 @@ export class Decentraland extends EventEmitter {
...
+      throw new Error('Linker-dapp responses an invalid scene sign')
```
Comment (gonpombo8) [inline src/lib/Decentraland.ts:252]:
```suggestion
      throw new Error('Linker DApp returned invalid scene sign')
```

File: src/lib/Decentraland.ts [hunk 8]
```diff
@@ -226,16 +232,44 @@ export class Decentraland extends EventEmitter {
...
     }
+    return response
+  }
 
-    return this.link(messageToSign)
+  async getIdentity(): Promise<LinkerResponseIdentity> {
```
Comment (gonpombo8) [inline src/lib/Decentraland.ts:257]: if this fn does the same as `getAddressAndSignature(message: string`, why no reuse it ?

```suggestion
  async getAddressAndSignature(message?: string, type: 'identity' | 'scene-deploy'): Promise<LinkerResponseIdentity> {
  
  }
```

File: src/lib/LinkerAPI.ts [hunk 9]
```diff
@@ -102,18 +98,37 @@ export class LinkerAPI extends EventEmitter {
...
+          estateRegistry: EstateRegistry,
+          debug: isDebug()
+        })
+        res.redirect(`/?${query}`)
+      }
+    })
+
+    this.app.get('/api/address_info', (req, res) => {
+      res.writeHead(200)
+      res.end()
```
Comment (gonpombo8) [inline src/lib/LinkerAPI.ts:129]:
```ts
res.sendStatus(200)
````

File: package.json [hunk 10]
```diff
@@ -69,7 +69,8 @@
     "ts-node": "^4.1.0"
   },
   "dependencies": {
-    "@dcl/linker-dapp": "^0.1.3",
+    "@dcl/builder-client": "^1.8.0",
+    "@dcl/linker-dapp": "^0.2.2-2090471827.commit-84b7063",
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply package.json]: linker-dapp ✅ 
common-schema 👎 yet

File: src/commands/deploy/deploySmartWearable.ts [hunk 11]
```diff
@@ -0,0 +1,234 @@
...
+    const results = await inquirer.prompt({
+      type: 'confirm',
+      name: 'continue',
+      message: warning(
+        `You haven't assigned a valid collection ID. Using collection can help to group wearables and test them. Do you want to assign one? It should already exist.`
+      )
+    })
+
+    if (results.continue) {
+      while (true) {
```
[the remark this answers is not in the document]
Comment (leanmendoza) [reply src/commands/deploy/deploySmartWearable.ts:69]: it ensures that the collectionId is valid (if you have a typo in your first try)

File: src/commands/deploy/deploySmartWearable.ts (hunk 3 above)
Comment (leanmendoza) [reply src/commands/deploy/deploySmartWearable.ts]: it has to be resolved by schemas side

File: src/commands/deploy/deploySmartWearable.ts (hunk 4 above)
Comment (leanmendoza) [reply src/commands/deploy/deploySmartWearable.ts]: done

File: src/lib/LinkerAPI.ts (hunk 9 above)
Comment (leanmendoza) [reply src/lib/LinkerAPI.ts]: removed because it never was used

File: src/lib/Decentraland.ts (hunk 8 above)
Comment (leanmendoza) [reply src/lib/Decentraland.ts:257]: almost

**Comment by Toromambo** [comment]
Yes but improve UX

**Pull request closed without merging**